modules/data-table/main/index.coffee

```diff
@@ -168,42 +222,56 @@ createAdvancedSearchView = (selection, dataTable, options) ->
       @append('div').class('hx-data-table-advanced-search-filter hx-section hx-input-group hx-input-group-full-width')
         .add(typePickerSel)
         .add(columnPickerSel)
-        .add(termInput)
-        .add(removeBtn)
+        .add(criteriaAnyPlaceholder)
+        .add(criteriaPickerSel)
+        .add(hx.div('hx-data-table-advanced-search-filter-input-container hx-input-group hx-no-pad hx-no-border')
+          .add(termInput)
+          .add(removeBtn))
         .node()
 
-  advancedSearchRowUpdate = ({term, column}, element, index) ->
+  advancedSearchRowUpdate = ({term, column, criteria}, element, index) ->
     filterRowSel = hx.select(element)
 
     validContext = if not term then 'negative' else undefined
 
-    filterRowSel.select('.hx-data-table-advanced-search-type')
-      .component().value(if index is 0 then 'or' else 'and')
+    filterRowSel.select('.hx-data-table-advanced-search-type').component()
+      .value(if index is 0 then 'or' else 'and')
+
+    trueColumn = column or 'any'
+    filterRowSel.select('.hx-data-table-advanced-search-column').component()
+      .value(trueColumn)
+
+    columnCriteria = columnOptionLookup(options, 'advancedSearchCriteria', column) || []
+    criteriaItems = if trueColumn is 'any' then ['contains'] else ['contains', advancedSearchCriteriaValidate(columnCriteria)...]
```

Comment:
I couldn't manage to get the `set` validation to work as expected because it treats `advancedSearchCriteria('something')` as `get value for column 'something'` so instead I've applied the validation logic when rendering the table - The value can be set to an invalid value but won't be rendered and will show a console warning